core/src/main/java/org/apache/carbondata/core/datastore/page/statistics/PrimitivePageStatsCollector.java

```diff
@@ -256,7 +256,7 @@ private int getDecimalCount(double value) {
   }
 
   private int getDecimalCount(float value) {
-    return getDecimalCount((double) value);
+    return getDecimalCount(Double.parseDouble(Float.toString(value)));
```

Comment:
In `AdaptiveDeltaFloatingCodec.java` line please remove line 323 to 328. As this code is added as a cover up to this issue. 

```
} else if (pageDataType == DataTypes.LONG) {
          int size = pageSize * longSizeInBytes;
          for (int i = 0; i < size; i += longSizeInBytes) {
            vector.putDouble(rowId++, (max - ByteUtil.toLongLittleEndian(pageData, i)) / factor);
          }
        }
```

Similarly in `AdaptiveFloatingCodec` 309 to 313

Reply:
removed